In [ ]:
################################### DOXX2TXT
#DOCX

In [1]:
import os
fileExt = r".docx"
fileDir = r"/Users/wpessoa/repositorios/NGE/words/"
print(os.path.exists(fileDir)) 
#fileDir = r"/Users/wpessoa/repositorios/Fluxo_Editorial/words/"
#fileDir= r'/Users/walnerpessoa/repositories/Fluxo_Editorial/words'
files_array = [_ for _ in os.listdir(fileDir) if _.endswith(fileExt)]
print(len(files_array))

True
35


In [2]:
import get_file_word as gf
import datetime
#fileDir = r"/Users/wpessoa/repositorios/Fluxo_Editorial/words/"
#fileDir= r'/Users/walnerpessoa/repositories/Fluxo_Editorial/words'
#fileExt = r".docx"

# NºID,Qtd_caracteres,Qtd_tabela,Qtd_image,data)
            
var=gf.get_info(fileDir,files_array)
#var2=gf.get_info(fileDir,fileExt,1)
#var3=gf.get_info(fileDir,fileExt,2)

print(var)
#print(var2)
#print(var3)


#gf.get_info(fileDir,fileExt)


[('233893', 216, 0, 1, datetime.date(2020, 2, 14)), ('235502', 32209, 7, 8, datetime.date(2020, 5, 27)), ('233694', 74718, 12, 67, datetime.date(2019, 5, 16)), ('233896', 92519, 50, 95, datetime.date(2020, 2, 11)), ('233899', 72272, 70, 141, datetime.date(2020, 2, 18)), ('235305', 106149, 81, 165, datetime.date(2020, 4, 20)), ('233903', 125108, 103, 179, datetime.date(2020, 3, 4)), ('235512', 83759, 115, 205, datetime.date(2020, 6, 8)), ('233904', 77964, 132, 209, datetime.date(2020, 3, 6)), ('233900', 75298, 132, 224, datetime.date(2020, 3, 2)), ('233892', 13373, 161, 224, datetime.date(2019, 6, 18)), ('233909', 90955, 168, 226, datetime.date(2020, 3, 11)), ('233867', 59270, 171, 237, datetime.date(2019, 12, 6)), ('233884', 213336, 239, 238, datetime.date(2020, 1, 20)), ('235353', 65306, 242, 240, datetime.date(2020, 5, 6)), ('235313', 144436, 251, 246, datetime.date(2020, 5, 6)), ('233885', 99723, 270, 255, datetime.date(2020, 2, 5)), ('233891', 16359, 275, 255, datetime.date(2020, 2

In [4]:
# retorna (NºID,Qtd_caracteres,Qtd_tabela,Qtd_image,data)

print(var[1])


('235502', 32209, 7, 8, datetime.date(2020, 5, 27))


In [ ]:
# Importar bibliotecas
from os import listdir
from os.path import isfile, join
import docx2txt
import docx
import os
import glob

from os import listdir
from os.path import isfile, join

In [ ]:
import glob
#targetPattern = r"/Users/wpessoa/repositorios/Fluxo_Editorial/words/*.docx"
targetPattern = r"/Users/walnerpessoa/repositorios/Fluxo_Editorial/words/*.docx"

#mypath='/Users/walnerpessoa/repositories/Fluxo_Editorial/words'
print(targetPattern)
files_array = glob.glob(targetPattern)
print(files_array)

In [ ]:
#fileDir = r"/Users/wpessoa/repositorios/Fluxo_Editorial/words/"
fileDir= r'/Users/walnerpessoa/repositories/Fluxo_Editorial/words'
fileExt = r".docx"
[_ for _ in os.listdir(fileDir) if _.endswith(fileExt)]

In [ ]:
# definir uma variável para o local onde estão os arquivos Word
#mypath="/Users/wpessoa/repositorios/Fluxo_Editorial/words/"
#mypath='/Users/walnerpessoa/repositories/Fluxo_Editorial/words'

# gerar vetor com ons nomes dos arquivos Word
onlyfiles = [f for f in listdir(fileDir) if isfile(join(fileDir, f))]

# verificar se existe o diretório
#os.path.exists('/Users/wpessoa/repositorios/Fluxo_Editorial/words')
os.path.exists(fileDir) 

In [ ]:
print(onlyfiles[0])
#docx2txt.process(onlyfiles[1])
doc = docx.Document(fileDir+'/'+onlyfiles[0])
#print(fileDir+'/'+onlyfiles[0])


In [ ]:
indice = 0
image_array=[]
table_chunks = []

# abrir conecção com o arquivo Word 
doc = docx.Document(fileDir+'/'+onlyfiles[indice])

# data criação
dt_doc = doc.core_properties.created
# ler em cada parágrafo dentro do arquivo Word
paragra= [p.text for p in doc.paragraphs]

# extai a quantidade de caracteres
caracteres = docx2txt.process(fileDir+'/'+onlyfiles[indice])

# gerar um array com tamanho de todas as imagens
for image in doc.inline_shapes:
    image_array.append([image.width, image.height])
    # print (image.width, image.height)
    
# gerar um array com todas as tabelas
for table in doc.tables:
    table_chunks.append(table)

#Saída dos dados
print('Nome do diretório: ',fileDir)
print('Nome do arquivo: ',onlyfiles[indice])
print("Data de Criação do Arquivo: ", dt_doc,"\n")
print('Quantidade de carcteres:',len(caracteres))
print("Quantidade de parágrafos: ", len(paragra))
print("Quantidade de imagens no vetor de imagens/gráficos: ",len(image_array))
print("Quantidade de Tabelas: ",len(table_chunks))


Nome do diretório:  /Users/wpessoa/repositorios/Fluxo_Editorial/words/
Nome do arquivo:  ID233867-06_12_2019_EFEITOS DA GESTÃO SOBRE A PRODUTIVIDADE - PARA PUBLICAÇÃO_Final_19122019_155125.docx
Quantidade de carcteres: 59270

Quantidade de parágrafos:  356
Quantidade de imagens no vetor de imagens:  11
Quantidade de Tabelas:  3
Data de Criação do Arquivo:  2019-12-06 13:09:00

In [ ]:
#####

In [ ]:
#teste com metodos da biblioteca Docx

print(doc.paragraphs[0].runs)
print(doc.paragraphs[0].text)
print(doc.paragraphs[0].runs[0].text)
print(doc.paragraphs[0].runs[0].style)
#print(doc.paragraphs[1].runs[0].text)
#print(doc.paragraphs[1].runs[0].style)
print(doc.core_properties.created)
print(doc.inline_shapes)
print(doc.inline_shapes[0].type)
print(doc.inline_shapes[0].height)
print(doc.inline_shapes[0].width)

#
